### PR TITLE
new builds/fix conflicts

### PR DESCRIPTION
**Pull request by Enjoy2Live**
- feat(RETprojects): working on like buttons
- Resolved merge conflict
- Attempting to get ideaId and related comments
- Getting ideaId
- Roman/tailwind-implemetation: tailwind config.
- Roman/tailiwing-implementation: Converting some css values.
- Worked on comment feed's visual design
- upgrade to storybookv7
- style sheet changes so far
- fix chromatic
- notes
- github workflow.
- Update Tokens
- renamed black token to blackk
- Update Tokens
- Update create-tokens.yml
- comment out storybook testing job
- Update custom.css
- extend tailwind config
- fix tokens have spaces
- Update Tokens
- Update create-tokens.yml
- fix padding tokens values
- fix border radius tokens values
- Update Tokens
- fix font size tokens values
- Update Tokens
- add all categories to tailwind config
- Update Tokens
- create dev command for tailwind classes
- added tailwind variants
- fix duplicated tw categories
- added gap values
- Update Tokens
- Working with the updated visual design further
- fix tailwind transpiling tokens wrongly
- moved fontFamily to the root
- Update Tokens
- fix tailwind transpiling tokens wrongly
- remove colors nested folder
- Refactor spacing tokens to have no folders
- fix issues with yarn not installing correctly
- fix reference errors
- Update Tokens
- upgrade storybook
- Revert "upgrade storybook"
- properly migrate to storybook v7.2
- WIP
- Debugging and updating
- Added second comment form at bottom of feed
- chore: add shadow tokens
- Update Tokens
- Making sure that new comments can be written
- Comparing the two versions of DisplayComments and WorkshoppingPage
- Debugging for comments state
- Making updates to DisplayComments
- Testing router
- completed frontend changes
- import the compiled tailwind.css file
- Got comments to display again
- added new color styles
- Update Tokens
- Displaying creation date of comments
- CSS changes to SaveIdea button
- final CSS changes for SaveIdea Button
- feat: added tabs component
- refactor: remove unsed props
- Formatted dates in comment feed
- WIP navigation
- fix: boxShadow transformation
- Update Tokens
- move brand colors to be at root level
- Update Tokens
- Update CODEOWNERS to have recruitment and user-profile teams
- Upgrade staging image to devlaunchers/platform-website:a7fea4d-202309220028
- variable to  store for savedCards.includes(id)
- Added idea_card in data for new comment
- Update CODEOWNERS to have the new @dev-launchers/dev-recruit team
- Upgrade staging image to devlaunchers/platform-website:e69f3a4-202309232150
- Cleaned up the comments and print statements
- Deleted second comment field
- Revised comment post body and deleted Kris user data
- fix(make idea card more functional/visually appealing)
- update ideaspace functionality
- fix(clean code)
- add snippets for creating components and stories
- mirror tailwind default screens to storybook
- first version of grid component
- target specific screens from sm up to xl
- Ordered comments by most recent
- Removed log statements
- feat: added new navigation options
- fix(capitalization on idea form)
- fix deps tree
- fix dropdown chevron img
- add tailwind to apps and finalize navbar
- match formik versions
- Upgrade staging image to devlaunchers/platform-website:9dd028b-202310312257
- fix: change query parameter for filtering projects to strapiv4 format
- fix navigation not redirecting to user profile
- Upgrade staging image to devlaunchers/platform-website:1ca3bf6-202311011838
- fix(global style issues)
- fix(remove ideaspace global style)
- Upgrade staging image to devlaunchers/platform-website:d19799f-202311042013
- re-adding the log out button
- fix(clean code and address build error)
- Delete .github/workflows/staging-storybook.yml
- Upgrade staging image to devlaunchers/platform-website:9c4e7c7-202311092326
- Upgrade staging image to devlaunchers/platform-website:e01a5f8-202311092342
- Edited comment feed to match Figma file more closely
- initial changes
- add gptbot staging app
- Upgrade staging image to devlaunchers/platform-website:99cb836-202311170428
- feat: add eslint bases for tailwind
- add tailwind eslint base rules to components lib
- adjust the linting based on our tailwind config
- fix linting errors that are autofixable
- fix remaining linting errors
- Removed commented-out code from SingleComment
- This adds state to dropdown options component, removing the persistant buttons bug
- delete tailwind.css
- stop tracking generated css
- generate css in CI
- Upgrade staging image to devlaunchers/platform-website:380484a-202311181952
- fix-lint className orders and unused classNames
- made backend changes to save ideacard
- updated project: router.query.slug as string
- fix tailwind intellisense
- warn for tailwindcss/no-arbitrary-value
- Toggle like button
- Upgrade staging image to devlaunchers/platform-website:19ded60-202311261905
- add the CSS compilation step to getting started section
- Update README.md
- map pages data to work with getOne controller
- hide images until API provides it
- fix(chungthuang): Remove the pop up on the landing page to recruit members
- Upgrade staging image to devlaunchers/platform-website:61cf245-202312012203
- revert changes made previously for getOne controller
- Revert "hide images until API provides it"
- Upgrade staging image to devlaunchers/platform-website:989a6a1-202312020034
- don't populate deep and populate * instead
- Upgrade staging image to devlaunchers/platform-website:0c3d688-202312020101
- fix: don't populate deep and populate * instead
- fix: made comments for user-profile team
- Upgrade staging image to devlaunchers/platform-website:cfd0630-202312020121
- Upgrade production image to devlaunchers/platform-website:2.12.2
- Revised timeSince(date) to return 1 year instead of 1 years, etc.
- fix(userData reference in idea card)
- fix: join page not rendering
- fix: remove duplicated header & footer this commit removes the extra header & footer from /page/* routes
- fix: bring back rainbow on nav bar
- Upgrade staging image to devlaunchers/platform-website:9424141-202312032221
- update dashboard page
- Upgrade production image to devlaunchers/platform-website:2.12.3
- Upgrade staging image to devlaunchers/platform-website:1f4e6a1-202312032306
- userData -> userData.id
- chore(chungthuang): Remove points, they are not used anymore
- fix: requests not having withCredentials param
- Upgrade staging image to devlaunchers/platform-website:c6613bc-202312101420
- fix: logout button not working on desktop
- fix: typescript errors
- fix: make withCredentials the default of every request
- Upgrade staging image to devlaunchers/platform-website:c5b1275-202312101639
- Upgrade staging image to devlaunchers/platform-website:af8c314-202312101644
- add user data to idea overview page
- Upgrade staging image to devlaunchers/platform-website:6e5ffd1-202312101855
- Upgrade production image to devlaunchers/platform-website:2.13.0
- Upgrade staging image to devlaunchers/platform-website:e4ad5ac-202312110037
- todd/tags (#1384)
- Draft
- Tailwind not working
- saved for now
- 4 states and check state added
- Extended React.html
- eslint
- removed efault and changed state var
- removed artifact and moved prop to index.ts
- added RadixCheckbox and moved tv out of jsx
- fixed default checked
- moved props into Checkbox for review
- input
- Extend checkboxprops
- removed omit
- omit checked not state
- Removed Global and yarn'd
- current changes
- try react-docgen instead of ts one
- Merged and cleaned up ready for PR
- removed props
- checkbox working
- Resolved review comments
- Checkbox focused issues
- Saving to share doc blocks
- added Controls
- removed unused
- removed argTypes
- need Omit in the end
- Removed comments, jsdoc code
- removed redunant tailwind
- Upgrade staging image to devlaunchers/platform-website:da952e2-202401192311
- Upgrade production image to devlaunchers/platform-website:2.14.0
- slow merge from suggestions merged
- Removed className
- Dev Recruiters app files deleted
- IdeaSpace components/common useless files deleted
- Site-projects components/common useless files deleted
- Website components/common useless files deleted and issue completed
- chore: move general mdx files to System dir
- docs(storybook): create page for colors in the system
- Addedc controls and description from jsdoc
- removed redunant comments
- removed className
- Wrap contents of states + figma container
- displaying docs on phones properly
- hack to show props in storybook controls
- update addon designs
- feat: allow devs to compare design to story pass in env var STORYBOOK_FIGMA_ACCESS_TOKEN and use @storybook/addon-designs
- fix ordering
- linked figma node to each story
- added ref and wrapper
- remove storybook-addon-designs
- fix error when STORYBOOK_FIGMA_ACCESS_TOKEN env var is not passed
- Upgrade staging image to devlaunchers/platform-website:38b624e-202401311908

### CODE REVIEW

File: apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js [hunk 1]
```diff
@@ -40,28 +45,27 @@ function IdeaCard({ cards, cardType }) {
   }, [tagContent]);
 
   const reactivateIdea = async () => {
-    cards['status'] = 'workshopping';
+    cards["status"] = "workshopping";
     setButtonContent(`WAIT`);
 
     try {
-      const res = await agent.Ideas.getIdea(cards.id, new URLSearchParams(`populate=*`));
+      const res = await axios.put(
```
Comment (dbradham) [inline apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js:52]: is this change necessary?

File: apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js [hunk 2]
```diff
@@ -134,8 +138,9 @@ function IdeaCard({ cards, cardType }) {
...
       )}
       <UpdateFailure />
+
     </atoms.Box>
-  );
+  )
 }
 
-export default IdeaCard;
+export default IdeaCard
```
Comment (dbradham) [inline apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js:146]: is this change necessary?

File: apps/ideaspace/src/components/common/SignInButton/SignInButton.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import {
   MenuButton
 } from "./StyledSignInButton";
 
-export default function SignInButton({redirectUrl=(process.env.NEXT_PUBLIC_FRONT_END_URL + "/users/me")}) {
+export default function SignInButton({redirectUrl="https://devlaunchers.org/users/me"}) {
```
Comment (dbradham) [inline apps/ideaspace/src/components/common/SignInButton/SignInButton.js:6]: is this change necessary? it will prevent developers and testers from having the expected sign in experience in staging/local environments.

File: apps/app/.env.staging [hunk 4]
```diff
@@ -1,7 +1,5 @@
 NEXT_PUBLIC_NAME=STAGING
-NEXT_PUBLIC_API_BASE_URL=https://apiv4-staging.devlaunchers.org
-NEXT_PUBLIC_API_URL=https://apiv4-staging.devlaunchers.org/api
-NEXT_PUBLIC_STRAPI_URL=https://apiv4-staging.devlaunchers.org/api
-NEXT_PUBLIC_GOOGLE_AUTH_URL=https://apiv4-staging.devlaunchers.org/api/connect/google
-NEXT_PUBLIC_DISCORD_AUTH_URL=https://discord.com/api/oauth2/authorize?client_id=815294711983112194&redirect_uri=https%3A%2F%2Fapi-staging.devlaunchers.org%2Fusers%2Fauth%2Fdiscord%2Fcallback&response_type=code&scope=identify
-NEXT_PUBLIC_FRONT_END_URL=https://staging.devlaunchers.org
```
Comment (dbradham) [inline apps/app/.env.staging:7]: is there any reason we are getting rid of the `NEXT_PUBLIC_FRONT_END_URL` variable?

File: apps/dev-recruiters/src/components/common/Header/AccountDropdown/AccountDropdown.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
 import React from "react";
 import Link from "next/link";
 import Logout from "@utils/Logout";
+import { env } from "@utils/EnvironmentVariables";
```
Comment (dbradham) [inline apps/dev-recruiters/src/components/common/Header/AccountDropdown/AccountDropdown.js:4]: is this change necessary?

File: apps/dev-recruiters/src/components/common/Header/AccountDropdown/AccountDropdown.js [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 import {
   AccountMenuDropdownItem,
   AccountMenuDropdownButton,
   MenuButton,
 } from "./StyledAccountDropdown";
-import { useUserDataContext } from "@devlaunchers/components/src/context/UserDataContext";
```
Comment (dbradham) [inline apps/dev-recruiters/src/components/common/Header/AccountDropdown/AccountDropdown.js:9]: it looks like this hook is still referenced in the component

File: apps/dev-recruiters/src/components/modules/FormPage/signUpForm.tsx [hunk 7]
```diff
@@ -1,8 +1,10 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Field, Form, Formik, FormikHelpers } from 'formik';
 import * as Yup from 'yup';
 import { useRouter } from 'next/router';
+import { useUserDataContext } from '../../../contexts/UserDataContext';
 import { NewApplicant } from '@devlaunchers/models/newApplicant';
+import { SkillLevel } from '@devlaunchers/models/level';
```
Comment (dbradham) [inline apps/dev-recruiters/src/components/modules/FormPage/signUpForm.tsx:7]: is this import used?

File: apps/ideaspace/src/components/modules/BrowseIdeas/BrowseIdeas.js [hunk 8]
```diff
@@ -69,9 +68,20 @@ function BrowseIdeas() {
     setCards(cardsClone);
   };
 
-  React.useEffect(async () => {
-    const ideaCards = cleanDataList(await agent.Ideas.get(
-      new URLSearchParams(`populate=*&pagination[pageSize]=1000`)));
+  React.useEffect(() => {
+    axios
+      .get(`${process.env.NEXT_PUBLIC_STRAPI_URL}/idea-cards`, {
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/BrowseIdeas/BrowseIdeas.js:73]: can this be reverted? We are making it a requirement within ideaspace to use the agent for all request to the backend.

File: apps/ideaspace/src/components/modules/EditIdea/EditIdea.js [hunk 9]
```diff
@@ -68,37 +67,42 @@ function EditIdea() {
...
-  };
 
-  useEffect(async () => {
     if (ideaId) {
-      const idea = cleanData(await agent.Ideas.getIdea(ideaId, new URLSearchParams("populate=*")));
-      if (!idea || !idea.id || idea.id == 0) {
-        setGetError(true);
-        return;
-      }
+      axios.get(`${process.env.NEXT_PUBLIC_STRAPI_URL}/idea-cards/${ideaId}`)
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/EditIdea/EditIdea.js:85]: Same request as above... is there any reason we are removing references to the agent?

File: apps/ideaspace/src/components/modules/EditIdea/EditIdea.js [hunk 10]
```diff
@@ -121,12 +125,15 @@ function EditIdea() {
     setSending(true);
 
     try {
-      const data = cleanData(await agent.Ideas.put(ideaId, values));
+      const res = await axios.put(
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/EditIdea/EditIdea.js:128]: same as above w/r/t agent usage

File: apps/ideaspace/src/components/modules/WorkshoppingPage/StyledComponents.js [hunk 11]
```diff
@@ -1,9 +1,13 @@
 import styled from "styled-components";
 
 export const Container = styled.div`
-  background-color: ${({ theme }) => theme.colors.GREYSCALE_OFF_WHITE};
+<<<<<<< HEAD
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/StyledComponents.js:4]: It looks like this conflict still needs to be resolved

File: apps/ideaspace/src/pages/_app.js [hunk 12]
```diff
@@ -4,6 +4,7 @@ import { ThemeProvider } from 'styled-components';
 import GlobalStyle from '@devlaunchers/components/src/styles/global';
 import Head from 'next/head';
 
+import { UserDataProvider } from '@devlaunchers/components/context/UserDataContext';
```
Comment (dbradham) [inline apps/ideaspace/src/pages/_app.js:7]: is this change necessary?

File: apps/ideaspace/src/utils/Logout.js [hunk 13]
```diff
@@ -1,5 +1,6 @@
 import Router from "next/router";
 import axios from "axios";
+import { env } from "./EnvironmentVariables";
```
Comment (dbradham) [inline apps/ideaspace/src/utils/Logout.js:3]: is this needed? It's not being used.

**Comment by dbradham** [comment]
@Enjoy2Live Is there any specific reason this PR is directed at `master` instead of following the process you have outlined here? https://github.com/dev-launchers/dev-launchers-platform/wiki

**Comment by Enjoy2Live** [comment]
@dbradham this PR by no means should be merged.
I'm helping community development I found out they're behind master by too many commits which created alot of conflicts that I'm trying to resolve without altering anyone's app.

I don't know what I should be doing to prevent this from happening? I've already mentioned in the docs the necessity of merging master to development/* branches every 2/4 weeks but some tech leads still are behind, it's frustrating to go through this conflicts resolution process.

**Comment by dbradham** [comment]
> @dbradham this PR by no means should be merged. I'm helping community development I found out they're behind master by too many commits which created alot of conflicts that I'm trying to resolve without altering anyone's app.
> 
> I don't know what I should be doing to prevent this from happening? I've already mentioned in the docs the necessity of merging master to development/* branches every 2/4 weeks but some tech leads still are behind, it's frustrating to go through this conflicts resolution process.

My suggestion would be to have the tech lead of community development solve these issues. If they are not adhering to the processes we have defined, they should not be rewarded by having you solve all of the problems they have created.

**Comment by Enjoy2Live** [comment]
> > @dbradham this PR by no means should be merged. I'm helping community development I found out they're behind master by too many commits which created alot of conflicts that I'm trying to resolve without altering anyone's app.
> > I don't know what I should be doing to prevent this from happening? I've already mentioned in the docs the necessity of merging master to development/* branches every 2/4 weeks but some tech leads still are behind, it's frustrating to go through this conflicts resolution process.
> 
> My suggestion would be to have the tech lead of community development solve these issues. If they are not adhering to the processes we have defined, they should not be rewarded by having you solve all of the problems they have created.

@dbradham I agree but they've stepped down unfortunately so it's either me or it's going to get worse, sorry to bother you with the notifications during the process!
